### DISCUSSION TITLE
Interfacing to GraphBLAS for faster sparse matrix operations, and support for graph algorithms

### DISCUSSION THREAD

**Discussion by DrTimothyAldenDavis** [josdejong/mathjs#3166] [Ideas]
Would there be any interest in an interface from math.js to GraphBLAS?  The GraphBLAS project provides a wide range of high performance sparse matrix computations and is used by MATLAB, Julia, FalkorDB and other projects for sparse computations and graph algorithms.  Details on the project are here: https://graphblas.org/ and I have a youtube video playlist here: https://www.youtube.com/playlist?list=PL5EvFKC69QIw3FqXj7UmiN1p-a-qYnZVK (see in particular the first 2 videos).

SuiteSparse:GraphBLAS is a large package (about 130K lines of C/C++) so it would be too daunting to try to translate it all into JavaScript.  It's Apache 2 licensed (see https://github.com/DrTimothyAldenDavis/GraphBLAS).

There might be others in the GraphBLAS.org community would be interested in a JavaScript interface to GraphBLAS.  I would be able to help with it, by providing input on the design of a JavaScript API.

**Comment by josdejong** [comment]
Thanks Timothy for sharing this!

If anyone is interesting in thinking through integrating GraphBLAS into mathjs or simply porting it to JS please let me know.

**Comment by DrTimothyAldenDavis** [comment]
Great!  I'll forward a link to this discussion to the GraphBLAS community.

-- Tim
